### ISSUE TITLE
Display errors + ignition status issue

### ISSUE THREAD

**Issue by Fleetracking** [vitalidze/traccar-web#973]
Hi 
I'm using last backend version 3.8 + GUI 3.7.
I want to report some display issues to improve GUI 

1) "@" symbols ?!
![image](https://cloud.githubusercontent.com/assets/17362734/20735057/68b184e6-b695-11e6-95b4-537ad48aa3bd.png)

![image](https://cloud.githubusercontent.com/assets/17362734/20735113/d25ffad0-b695-11e6-8666-953564a8672e.png)

![image](https://cloud.githubusercontent.com/assets/17362734/20735147/23b3b71e-b696-11e6-8d4b-3da0bc632a7a.png)

2) "Unknown command" ?!
I suggest to optimize commands in function of protocol (exp: no  command to view for osmand protocol) also view an ack if command success.
![image](https://cloud.githubusercontent.com/assets/17362734/20735082/a43e07b4-b695-11e6-8cf8-d0f374d3eabb.png)

3) "distance" and "total distance" attributes viewed even if "show odometer" is desactivated at device conf window
![image](https://cloud.githubusercontent.com/assets/17362734/20735461/3576c700-b698-11e6-99e8-b63db2a352ab.png)

4) mismatch status (offline status with a red square in map)
![image](https://cloud.githubusercontent.com/assets/17362734/20735606/0cfd22c8-b699-11e6-8e3a-784279bba8c0.png)

5) mismatch "ignition" status (map report ignition at true status even if car is really stopped speed=0)
![image](https://cloud.githubusercontent.com/assets/17362734/20735962/76013384-b69b-11e6-8906-a72fbbdfd3e7.png)

**Comment by vitalidze** [comment]
1. and 2. are caused by an old version of web UI cached in your web browser. You need to clean up it's cache. This step is present in installation instructions.

> 3. "distance" and "total distance" attributes viewed even if "show odometer" is desactivated at device conf window

These are parameters (or "sensors" values) coming from your device. It has nothing to do with "odometer". You can hide them with [sensors](http://traccar.litvak.su/features/sensors.html) functionality.

> 4. mismatch status (offline status with a red square in map)

Looks like you don't understand the meaning of those two statuses. "Offline" means that your device does not send any data for some period of time (configurable in device profile). "Stopped" (or red square) means that your device is not moving. So this means that when your device last sent it's location it was not moving.

> 5. mismatch "ignition" status (map report ignition at true status even if car is really stopped speed=0)

This is what your device is reporting. It has nothing to do with web UI.

**Comment by Fleetracking** [comment]
Dear @vitalidze
Thank you first for your answers.
1 & 2) Problem resolved after reclearing browser cach.
Maybe customizing supported commands by protocol will be a futur enhancement

3) No sensor is configured for this device (it's a smartphone using Client Android App)

4) The snapshot is for a smartphone at offline status (no Data connection) 2h4m ago (configured timeout is 900s), so it's at the same time at "offline" status and also at "stopped" status because it's the last data available at the server (anyway for osmand protocol reported speed=0  i think).
So, can i conclude that at this case "stopped" status has more priority to report at GUI?

5) Maybe it's a backend interpretation's issue of H02 protocol, because i remarked that ignition status is always inversed.

Best regards

**Comment by vitalidze** [comment]
> Maybe customizing supported commands by protocol will be a futur enhancement

Agreed.

> No sensor is configured for this device (it's a smartphone using Client Android App)

Then you must have enabled distance calculation on the "backend" side with the `distance.enable` parameter in config file.

> So, can i conclude that at this case "stopped" status has more priority to report at GUI?

I don't understand how did you come to this conclusion. "Stopped" status has more priority over what?

> Maybe it's a backend interpretation's issue of H02 protocol, because i remarked that ignition status is always inversed.

Yes, it looks like this.

**Comment by Fleetracking** [comment]
Hi 
- I checked my "backend" config files "traccar.xml" and "default.xml", "distance.enable" parameter is not configured.

- Regarding device status my question is : when a device is disconnected from the server (no data received during configured Timeout) and at the same time device is in idle status (Speed < Idle_threshold during configured minimum period), what's the status reported by GUI ?

- regarding "ignition" issue, i reported my question at "backend" [forum](https://www.traccar.org/forums/topic/ignition-status-issue-h02-protocol/).

BR

**Comment by vitalidze** [comment]
> I checked my "backend" config files "traccar.xml" and "default.xml", "distance.enable" parameter is not configured.

Then maybe you had it configured at some point. Anyway, it has nothing to do with the web UI, this is what you have in your database. Odometer calculation on the Web UI side is done differently and does not invoke any sensor.

> Regarding device status my question is : when a device is disconnected from the server (no data received during configured Timeout) and at the same time device is in idle status (Speed < Idle_threshold during configured minimum period), what's the status reported by GUI ?

The "Offline" status is a parallel status with "Idle"/"Moving". Same applies to the "Online" status. So device may be moving and online, and can be moving and offline. There is no priority, system just reports what it received from device and shows you the status based on the settings.

**Comment by Fleetracking** [comment]
Hi everybody
Any idea regarding "distance" and "total distance" attributes?
These attributes are a part of "other" column at mysql DB (exp: {"battery":"38.0","ip":"52.139.121.75","**distance**":0.0,"**totalDistance**":399493.49}).
BR

**Comment by vitalidze** [comment]
I am out of ideas. I think it's worth asking same question on the ["backend" issue tracker](https://github.com/tananaev/traccar/issues).

**Comment by Fleetracking** [comment]
After asking question on backend forum, the answer is : 
Distance calculation is always enabled and can't be disabled and `distance.enable` has no effect since 3.7 version !!
Do you have a suggestion how to customize which "attribute" to show for which Device.
BR

**Comment by vitalidze** [comment]
Yes, as I have already said you can hide them with [sensors functionality](http://traccar.litvak.su/features/sensors.html)

**Comment by Fleetracking** [comment]
Hi
I feel that i begin to loop :) 
As mentioned no sensor is configured for these devises (smart phones), so why these "distance" attributes are viewed at GUI?
You answered me that i have "distance.enable" enabled at my confs files. 
After check, "distance.parameter" isn't configured ...
BR

**Comment by vitalidze** [comment]
All sensors are displayed by default unless you configure them to be hidden. You need to read the article to figure out how to do it.

**Comment by Fleetracking** [comment]
Thank you, finally attributes hiden by:
- First: Creat it (distance & totalDistance)
- Secondo: Hide it
But it's not indicated that all sensors are displayed by default if not added for some device.
Anyway thank you very much for your help and effort.
BR

**Comment by vitalidze** [comment]
> But it's not indicated that all sensors are displayed by default if not added for some device.

It's obvious from my point of view. This can be figured out just by looking into any position received from your device.

**Comment by sankalpask** [comment]
Hi,
had the same issue with distance & totalDistance. 
Is there any way to hide these parameters from all devices? 
(It's taking too much time if I have to create & hide sensors for all my devices)
Thanks.

**Comment by vitalidze** [comment]
@sankalpask There is no way to hide it from all devices, except configuring sensors. You can copy sensors between devices. There is a drop-down box and a button at the top of device editing screen. Another way is to copy them with some SQL query.

**Comment by sankalpask** [comment]
Thanks. I'll try this.
